server/apps/postgres-app/src/main/java/org/apache/james/PostgresJamesServerMain.java

```diff
@@ -170,6 +184,17 @@ private static List<Module> chooseBlobStoreModules(PostgresJamesConfiguration co
         return builder.build();
     }
 
+    public static List<Module> chooseTaskManagerModules(PostgresJamesConfiguration configuration) {
+        switch (configuration.eventBusImpl()) {
+            case IN_MEMORY:
+                return List.of(new TaskManagerModule());
+            case RABBITMQ:
+                return List.of(new DistributedTaskManagerModule(), new DistributedTaskSerializationModule());
```

Comment:
This `DistributedTaskSerializationModule` causes Postgres Distributed IT to fail:

```java
1) [Guice/CanNotProxyClass]: Tried proxying JsonTaskSerializer to support a circular dependency, but it is not an interface.
  at DistributedTaskSerializationModule.taskCancelRequestedSerialization(DistributedTaskSerializationModule.java:71)
      \_ for 1st parameter
  at DistributedTaskSerializationModule.taskCancelRequestedSerialization(DistributedTaskSerializationModule.java:71)
      \_ installed by: Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> Modules$OverrideModule -> DistributedTaskSerializationModule
  while locating EventDTOModule<? extends Event, ? extends EventDTO> annotated with @Element(setName=,uniqueId=350, type=MULTIBINDER, keyType=)
  at JsonEventSerializer.<init>(JsonEventSerializer.scala:64)
      \_ for 1st parameter modules
  at PostgresEventStoreDAO.<init>(PostgresEventStoreDAO.java:55)
      \_ for 2nd parameter
  at PostgresEventStore.<init>(PostgresEventStore.java:47)
      \_ for 1st parameter
```

I do not understand why this is not an issue with the `distributed-app` yet (I tried to do the same)...

Reply:
Maybe order? 

I had a weird thing in mine too https://github.com/linagora/tmail-backend/pull/989 if I put JMAPEventBusModule in the rabbitmq bloc like here I was having injection issues but somehow in chooseJmapModule it was ok... Could not figure out why

Reply:
thanks, let's me have a look

Reply:
I am running out of ideas why it worked for the distributed-app (https://github.com/apache/james-project/pull/1208#issuecomment-1264297916)... Other insights may help with this.

So I would disable `DistributedTaskSerializationModule` for now and handle the circular issue in another ticket.

Likely I would propose refactoring `JsonTaskSerializer` into an interface...

Reply:
cf https://github.com/linagora/james-project/issues/5150